### PR TITLE
v1 breaking changes 

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR

## Description 
Merges the "v1 breaking changes" branch into main. This branch was constructed from the following PRs:
 * #616 
 * #617 
 * #618 
 * #620 
 * #621 
 * #622 
 * #625 
 * #625 
 * #628
 * #630 

### PR Type:
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

**Comment by bfolie** [comment]
e2es against development: https://jenkins.corp.citrine.io/job/nextgen-ITs/17092/
e2es against stage: https://jenkins.corp.citrine.io/job/nextgen-ITs/17093/

**Review by kroenlein — APPROVED** [review]
It's been a heavy lift getting to this point.  Thanks for all the hard work; looks great!

File: src/citrine/__version__.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '0.133.5'
+__version__ = '0.134.0'
```
Comment (talltea) [inline src/citrine/__version__.py:1]: should this be 1.0.0?

Comment (bfolie) [reply src/citrine/__version__.py:1]: Not yet. We've got a few more small things to do, but they're not breaking. So this is like a "soft release": if a user gets their code up to date with this version, then they are ready for v1.

**Review by maxhutch — APPROVED** [review]
Excellent work @bfolie and @talltea !

**Comment by talltea** [comment]
Updated the docs with kwargs in that last commit

File: src/citrine/informatics/workflows/design_workflow.py [hunk 2]
```diff
@@ -32,86 +29,39 @@ class DesignWorkflow(Resource['DesignWorkflow'], Workflow, AIResourceMetadata):
...
-    description = properties.Optional(properties.String, 'description')
-
     design_space_id = properties.UUID('design_space_id')
     processor_id = properties.Optional(properties.UUID, 'processor_id')
     predictor_id = properties.UUID('predictor_id')
 
     status_description = properties.String('status_description', serializable=False)
     """:str: more detailed description of the workflow's status"""
 
+    # TODO: I think that module_type is a vestige of when workflows were accessed through a
```
Comment (talltea) [inline src/citrine/informatics/workflows/design_workflow.py:39]: yup, module type isn't necessary any more

File: src/citrine/resources/report.py [hunk 3]
```diff
@@ -22,9 +22,9 @@ def __init__(self, project_id: UUID, session: Session):
         self.project_id = project_id
         self.session = session
 
-    def get(self, module_id: UUID) -> Report:
+    def get(self, uid: UUID) -> Report:
```
Comment (talltea) [inline src/citrine/resources/report.py:25]: this snuck in, removing

**Comment by bfolie** [comment]
e2es against development passed. Against stage there was an unrelated, transient error, so I re-ran and they passed: https://jenkins.corp.citrine.io/job/nextgen-ITs/17098/

**Review by talltea — APPROVED** [review]
This is amazing, @bfolie!!!

**Comment by bfolie** [comment]
Tatiana's 2 recent commits look good to me (so everything has been reviewed by at least one other person).
Since it involved one tiny non-documentation change, I'm re-running the e2es to be super-duper-extra-careful. Once they finish then I will merge. https://jenkins.corp.citrine.io/job/nextgen-ITs/17104/

**Comment by bfolie** [comment]
The e2es had 2 failures that looked transient and unrelated. I ran again to be safe, and it passed (https://jenkins.corp.citrine.io/job/nextgen-ITs/17107/). Merging 😰

**Pull request merged**